### ISSUE TITLE
Clean up RSI(raft) comments

### ISSUE THREAD

**Issue by timmaxw** [rethinkdb/rethinkdb#3896]
There are a bunch of comments in the codebase of the form:
```c++
// RSI(raft): <thing that needs to get fixed before Raft ships>
```
Before we ship Raft we should go through and fix these comments, turn them into issues, or whatever.

(Many of them are redundant with issues currently in the Raft milestone, so we shouldn't do this until all those issues have been fixed.)

**Comment by timmaxw** [comment]
I fixed most of these in CR 2962. The remaining ones are related to open GitHub issues.

**Comment by timmaxw** [comment]
Changes were merged in 4f36e52d3133a3ef5ecf79159aa87022605d067f.

Let's leave this issue open to remember to take out the last few `RSI(raft)` comments.

**Comment by VeXocide** [comment]
The only comments left are related to metadata migration which is tracked by #3881, closing this.
